File: basejava/src/main/java/self/yue/basejava/vipe/CommonView.java
package self.yue.basejava.vipe;

import android.support.annotation.StringRes;

/**
 * Created by dongc on 10/21/2017.
 */

public interface CommonView {
  void showMessage(String message);

  void showMessage(@StringRes int messageResId);

  void showLoadingDialog(String loadingMessage);

  void dismissLoadingDialog();
}

File: basekotlin/src/main/java/self/yue/basekotlin/vipe/CommonView.kt
package self.yue.basekotlin.vipe

import android.content.Context
import android.support.annotation.StringRes

/**
 * Created by dongc on 10/20/2017.
 */
interface CommonView {
  fun showMessage(message: String)

  fun showMessage(@StringRes messageResId: Int)

  fun showLoadingProgress(context: Context)

  fun dismissLoadingProgress()
}